src/wallet/mod.rs

```diff
@@ -316,6 +316,51 @@ where
         }
     }
 
+    /// Ensures that there are at least `max_addresses` or `CACHE_ADDR_BATCH_SIZE` addresses
+    /// cached in the database if the descriptor is derivable, or 1 address if it is not.
+    /// Will return `Ok(true)` if there are new addresses generated (either external or internal),
+    /// and `Ok(false)` if all the required addresses are already cached. This function is used internally
+    /// by [`Wallet::sync`] in wallets that have a [`Blockchain`] client, and is useful for direct
+    /// use in offline wallets where [`Wallet::sync`] is unavailable but addresses need to be loaded
+    /// to do things like check [`Wallet::is_mine`] on transaction output scripts.
+    pub fn ensure_addresses_cached(&self, max_addresses: Option<u32>) -> Result<bool, Error> {
+        let mut run_setup = false;
```

Comment:
```suggestion
        let mut new_addresses_cached = false;
```

`run_setup` doesn't make sense.